lib/ResourceSet.h

```diff
@@ -32,60 +33,57 @@ namespace Res
 	};
 
 	//class to be representing a vector of resource
-	class ResourceSet : public std::vector<int>
+	class ResourceSet : public std::array<int, GameConstants::RESOURCE_QUANTITY>

```

Comment:
Perhaps replace inheritance with composition?
Instead of inheriting from container include it as member and provide required interface where necessary, e.g begin()/end()/operator[]

I would also keep vector as before - eventually we might want to allow new resources & keeps save format

Reply:
> eventually we might want to allow new resources
I don't see it, we already have Mithril which is unused. Also it may be difficult to handle more resources in GUI.
Overall the argumentation is similiar to adding more than 2 levels to a map.

Reply:
more than 2 levels of a map is on my list as important feature that should be eventually added

Reply:
> I don't see it, we already have Mithril which is unused. Also it may be difficult to handle more resources in GUI. Overall the argumentation is similiar to adding more than 2 levels to a map.

Maybe true, but we should not *increase* difficulty in implementing it.

Best approach here would be to hide this implementation detail in internal container and only provide public interface. (meaning - remove inheritance)
External code should not be aware whether number of resources is fixed or not.

And inheriting from standard containers is already rather suspicious code IMO.

Reply:
Done as composition, but as array.